tools/c7n_azure/c7n_azure/resources/vm.py

```diff
@@ -23,4 +24,21 @@ class resource_type(object):
         client = 'ComputeManagementClient'
         enum_spec = ('virtual_machines', 'list_all')
         id = 'id'
-        name = 'name'
\ No newline at end of file
+        name = 'name'
+        default_report_fields = (
+            'name',
+            'location',
+            'properties.hardwareProfile.vmSize',
+        )
+
+@VirtualMachine.filter_registry.register('instance-view')
+class InstanceViewFilter(ValueFilter):
+    schema = type_schema('instance-view', rinherit=ValueFilter.schema)
+
+    def __call__(self, i):
+        if 'instanceView' not in i:
+            client = self.manager.get_client()
+            instance = client.virtual_machines.get(i['id'].split('/')[4], i['name'], expand='instanceview').instance_view
+            i['instanceView'] = instance.serialize()
```

Comment:
we typically try to annotate those with a prefix, so that the serialized representation for object storage/disk is canonical with markups from additional lookups clearly annotated. also given the usage/context  we would typically call the super in this context with the instanceview instead of the instance so the user can specify matches with root of keys local to the instance view.

Reply:
So in this case I suspected this might be ideal, because this is actually the schema you would get if you did a `get` on a VM.  It is just the specific enumerator we use that omits this field for performance reasons (although the SDK is working on fixing it).